src/change_stream.ts

```diff
@@ -304,14 +295,86 @@ export class ChangeStream extends EventEmitter {
       const cursor = this.cursor;
 
       return cursor.close(err => {
-        ['data', 'close', 'end', 'error'].forEach(event => cursor.removeAllListeners(event));
+        this[kStream]?.cleanupListeners();
         this.cursor = undefined;
 
         return cb(err);
       });
     });
   }
 
+  /**
+   * Return a modified Readable stream including a possible transform method.
+   * @throws MongoError if this.cursor is undefined
+   */
+  stream(options?: StreamOptions): ChangeStreamStream {
+    if (this[kStream] == null) {
+      this[kStream] = new ChangeStreamStream(this, options);
+    }
+    return this[kStream] as ChangeStreamStream;
+  }
+
+  repipeStreams(changeStreamCursor: ChangeStreamCursor): void {
+    const stream = this[kStream];
+    if (stream == null) return;
+    if (stream.pipeDestinations) {
+      const cursorStream = changeStreamCursor.stream(stream.streamOptions);
+      for (const pipeDestination of stream.pipeDestinations) {
+        cursorStream.pipe(pipeDestination);
+      }
+    }
+  }
+}
+
+class ChangeStreamStream extends EventEmitter {
+  changeStream: ChangeStream;
+  pipeDestinations: Writable[] = [];
+  cursor: ChangeStreamCursor;
+  streamOptions: StreamOptions;
+  cursorStream: CursorStream;
```

Comment:
so this is kind of what I was talking about the other day, I think we actually want to make `ChangeStreamStream extends Readable` or at least `extends CursorStream`, so that we can be in control over the resumability of the stream. Similarly to how we want people to only write `changeStream.on('change', ...)` one time and not think about resuming, we want people doing `changeStream.stream().pipe(process.stdout)` to never have to write resume logic themselves.